backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/rabbitmqhttp/process/C_BPartner_SyncTo_RabbitMQ_HTTP.java

```diff
@@ -0,0 +1,124 @@
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.externalsystem.rabbitmqhttp.process;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import de.metas.bpartner.BPartnerId;
+import de.metas.externalsystem.ExternalSystemConfigRepo;
+import de.metas.externalsystem.ExternalSystemParentConfig;
+import de.metas.externalsystem.ExternalSystemType;
+import de.metas.externalsystem.rabbitmqhttp.ExternalSystemRabbitMQConfigId;
+import de.metas.externalsystem.rabbitmqhttp.RabbitMQExternalSystemService;
+import de.metas.process.IProcessDefaultParameter;
+import de.metas.process.IProcessDefaultParametersProvider;
+import de.metas.process.IProcessPrecondition;
+import de.metas.process.IProcessPreconditionsContext;
+import de.metas.process.JavaProcess;
+import de.metas.process.Param;
+import de.metas.process.ProcessPreconditionsResolution;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBuilder;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_C_BPartner;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Set;
+
+public class C_BPartner_SyncTo_RabbitMQ_HTTP extends JavaProcess implements IProcessPrecondition, IProcessDefaultParametersProvider
+{
+	private final RabbitMQExternalSystemService rabbitMQExternalSystemService = SpringContextHolder.instance.getBean(RabbitMQExternalSystemService.class);
+	private final ExternalSystemConfigRepo externalSystemConfigDAO = SpringContextHolder.instance.getBean(ExternalSystemConfigRepo.class);
+
+	private static final String PARAM_EXTERNAL_SYSTEM_CONFIG_RABBITMQ_HTTP_ID = "ExternalSystem_Config_RabbitMQ_HTTP_ID";
+	@Param(parameterName = PARAM_EXTERNAL_SYSTEM_CONFIG_RABBITMQ_HTTP_ID)
+	private int externalSystemConfigRabbitMQId;
+
+	@Nullable
+	@Override
+	public Object getParameterDefaultValue(final IProcessDefaultParameter parameter)
+	{
+		if (PARAM_EXTERNAL_SYSTEM_CONFIG_RABBITMQ_HTTP_ID.equals(parameter.getColumnName()))
+		{
+			final ImmutableList<ExternalSystemParentConfig> activeConfigs = externalSystemConfigDAO.getAllByType(ExternalSystemType.RabbitMQ)
+					.stream()
+					.filter(ExternalSystemParentConfig::getIsActive)
+					.collect(ImmutableList.toImmutableList());
+
+			return activeConfigs.size() == 1
+					? activeConfigs.get(0).getChildConfig().getId().getRepoId()
+					: IProcessDefaultParametersProvider.DEFAULT_VALUE_NOTAVAILABLE;
+		}
+		return IProcessDefaultParametersProvider.DEFAULT_VALUE_NOTAVAILABLE;
+	}
+
+	@Override
+	public ProcessPreconditionsResolution checkPreconditionsApplicable(final @NonNull IProcessPreconditionsContext context)
+	{
+		if (context.isNoSelection())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+		}
+
+		final ImmutableList<ExternalSystemParentConfig> configs = externalSystemConfigDAO.getAllByType(ExternalSystemType.RabbitMQ)
```

Comment:
consider introducing a method like ExternalSystemConfigRepo.isRabbitMQActive() which does what this code snippet does.

Reply:
done